Tests/SwiftProtobufTests/Test_JSONDecodingOptions.swift

```diff
@@ -76,6 +76,9 @@ class Test_JSONDecodingOptions: XCTestCase {
             (true, "{\"unknown\":  7.308e+8}"),
             (true, "{\"unknown\": \"hi!\"}"),
             (true, "{\"unknown\": []}"),
+            (true, "{\"unknown\": [[]]}"),
+            (true, "{\"unknown\": [[[]]]}"),
+            (true, "{\"unknown\": [[],[[]],[[[]],[]],[],7]}"),
```

Comment:
Maybe throw in a case with nested values?  `[[[1, 2]]]`

And a cases with values before/after the nested arrays: `[[1,[[]],3]]`

And do you want a test some place to ensure malformed nested is trapped and doesn't crash?  `[[[1,]]]`, `[[[,1]]]`, `[[[],]]`, `[[,[]]]`?

Reply:
I added a bunch more cases here, including a bunch that fail parsing.